nixos/modules/services/networking/networkmanager.nix

```diff
@@ -379,6 +380,56 @@ in {
           for more details.
         '';
       };
+      ensureProfiles = {
+        profiles = mkOption {
+          type = types.attrsOf ini.type;
+          default = { };
+          example = {
+            home-wifi = {
+              connection = {
+                id = "home-wifi";
+                type = "wifi";
+                permissions = "";
+              };
+              wifi = {
+                mac-address-blacklist = "";
+                mode = "infrastructure";
+                ssid = "Home Wi-Fi";
+              };
+              wifi-security = {
+                auth-alg = "open";
+                key-mgmt = "wpa-psk";
+                psk = "$HOME_WIFI_PASSWORD";
+              };
+              ipv4 = {
+                dns-search = "";
+                method = "auto";
+              };
+              ipv6 = {
+                addr-gen-mode = "stable-privacy";
+                dns-search = "";
+                method = "auto";
+              };
+            };
+          };
+          description = lib.mdDoc ''
+            Declaratively define NetworkManager profiles. You can find information about the generated file format [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/assembly_networkmanager-connection-profiles-in-keyfile-format_configuring-and-managing-networking).
+            You current profiles which are most likley stored in `/etc/NetworkManager/system-connections` and there is [a tool](https://github.com/janik-haag/nm2nix) to convert them to the needed nix code.
+            If you add a new adhoc connection via a gui or nmtui or anyting similiar it should just work together with the declarativ ones.
+            And If you edit a declrativ profile NM will move it to the persitent storage and treat it like a adhoc one,
+            but there will be two profiles as soon as the systemd unit from this option runs again which can be confusing since NM tools will start displaying two profiles with the same name and probalby a bit diffrent settings deppending on what you edited.
```

Comment:
```suggestion
            but there will be two profiles as soon as the systemd unit from this option runs again which can be confusing since NetworkManager tools will start displaying two profiles with the same name and probably a bit diffrent settings depending on what you edited.
```